### DISCUSSION TITLE
Mobile Specific Views / Device Detection

### DISCUSSION THREAD

**Discussion by tysonnero** [Dotnet-Boxed/Templates#110] [Q&A]
In the .NET Core docs there is a page titled "Building Mobile Specific Views" but is under construction: https://docs.asp.net/en/latest/mvc/views/mobile.html.

Does anyone have some insight on building mobile views or successfully doing device detection?

**Comment by RehanSaeed** [comment]
I think I saw a thread (probably was you) asking this question. The obvious answer is that you probably shouldn't do this as device detection is ridiculously difficult, all browsers half lie about who they are because websites are doing device detection wrongly.

Instead you should use CSS media queries to adjust to the page based on viewport size. That said, if you have a **really** good reason to do this, there may be some project somewhere where you can find a big database of user agents or something.
